src/types/box.js

```diff
@@ -18,12 +18,11 @@ export default class BoxAnnotation extends Element {
   }
 
   drawLabel(ctx) {
-    const {x, y, width, height, options} = this;
-    const labelOpts = options.label;
-    labelOpts.borderWidth = options.borderWidth;
+    const {x, y, width, height, options: {label: labelOpts}} = this;
+    const halfBorder = labelOpts.borderWidth / 2;
     ctx.save();
     ctx.beginPath();
-    ctx.rect(x + labelOpts.borderWidth / 2, y + labelOpts.borderWidth / 2, width - labelOpts.borderWidth, height - labelOpts.borderWidth);
+    ctx.rect(x + halfBorder, y + halfBorder, width - halfBorder, height - halfBorder);
```

Comment:
Was the change to use `width - halfBorder` intended?

Reply:
good catch

The version requirement is due to https://github.com/chartjs/Chart.js/pull/9969 and is only required when annotations are passed as an array. There are already similar issues in current version and arrays with chart.js < 3.7.

So for proper functionality 3.7 should be used with any version of this plugin, when using annotations as array.

Reply:
Fixed, the clip did not account for padding either, which I think it should (applied).